js/ui/handler/drag_pan.js

```diff
@@ -31,47 +30,73 @@ DragPan.prototype = {
     },
 
     _onDown: function (e) {
-        this._startPos = this._pos = DOM.mousePos(this._el, e);
+        if (this.active) return;
+
+        if (e.touches && e.touches.length === 1) {
+            document.addEventListener('touchmove', this._onMove);
+            document.addEventListener('touchend', this._onTouchEnd);
+        } else if (e.button === 0) {
+            document.addEventListener('mousemove', this._onMove);
+            document.addEventListener('mouseup', this._onMouseUp);
+        } else {
+            return;
+        }
 
+        this.active = false;
+        this._startPos = this._pos = DOM.mousePos(this._el, e);
         this._inertia = [[Date.now(), this._pos]];
-
-        if (!e.touches) {
-            document.addEventListener('mousemove', this._onMove, false);
-            document.addEventListener('mouseup', this._onMouseUp, false);
-
-        } else if (e.touches.length === 1) {
-            document.addEventListener('touchmove', this._onMove, false);
-            document.addEventListener('touchend', this._onTouchEnd, false);
-        }
     },
 
     _onMove: function (e) {
         var map = this._map;
-        if (map.boxZoom.active || map.dragRotate.active || (e.touches && e.touches.length > 1)) return;
 
-        this.active = true;
+        if (map.boxZoom && map.boxZoom.active) return;
+        if (map.dragRotate && map.dragRotate.active) return;
+        if (e.touches && e.touches.length > 1) return;
+        if (!e.touches && e.buttons & 1 === 0) return;
+
+        if (!this.active) {
+            this.active = true;
+            this._fireEvent('dragstart', e);
+            this._fireEvent('movestart', e);
+        }
 
         var pos = DOM.mousePos(this._el, e),
             inertia = this._inertia,
             now = Date.now();
 
         inertia.push([now, pos]);
-        while (inertia.length > 2 && now - inertia[0][0] > 50) inertia.shift();
+        while (inertia.length > 1 && now - inertia[0][0] > 50) inertia.shift();
 
         map.stop();
         map.transform.setLocationAtPoint(map.transform.pointLocation(this._pos), pos);
-        map.fire('move');
+
+        this._fireEvent('drag', e);
+        this._fireEvent('move', e);
 
         this._pos = pos;
 
         e.preventDefault();
     },
 
-    _onUp: function () {
-        var inertia = this._inertia;
+    _onUp: function (e) {
+        var map = this._map;
+
+        if (!this.active) return;
+        if (map.boxZoom && map.boxZoom.active) return;
+        if (map.dragRotate && map.dragRotate.active) return;
+        if (e.touches && e.touches.length > 1) return;
+        if (!e.touches && e.button !== 0) return;
```

Comment:
this is repeated two times so maybe worth separating into a method that returns boolean